### DISCUSSION TITLE
Support for Horizon 7.4AT-02?

### DISCUSSION THREAD

**Discussion by MrPicklesRocks** [cagnulein/qdomyos-zwift#2196] [General]
Hi - I'm struggling to get workout data from our Horizon 7.4AT-02 treadmill into Apple Health. I just found that the treadmill does not support their newer app (atzone) and the older app (AFG Pro) doesn't communicate with Apple Health.

I was going to look into QZ but want to be sure this will work with my equipment. I'm just trying to collect time, distance, pace, calories, etc.

Will QZ work for this and would I still need to use Zwift or other? I don't really need to control the machine from the app.

Thanks!

**Comment by cagnulein** [comment]
Hi! no you don't need to use zwift at all. yes qz can write all the metric
to apple health with your apple watch!

to be sure that your treadmill is compatible, Install the free app
nrfconnect From the store and attached a screenshot of the treadmill in the
app. Let me know

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno ven 8 mar 2024 alle 18:20 MrPicklesRocks ***@***.***>
ha scritto:

**Comment by MrPicklesRocks** [comment]
Here’s what I found in the log:

Scanner On.
Device Scanned.
Device Appearance switched from Generic to Fitness.
Connected.
Discovered FFF0, Fitness Machine, and FEE7 Services.
Discovered Characteristics FFF1, FFF2, FFF3, FFF4, and FFF5 for Service FFF0.
Discovered Characteristics Fitness Machine Feature, Indoor Bike Data, Treadmill Data, Step Climber Data, Cross Trainer Data, Fitness Machine Control Point, Supported Speed Range, Supported Inclination Range, Supported Resistance Level Range, Supported Power Range, Supported Heart Rate Range, Fitness Machine Status, Training Status, Rower Data, and Stair Climber Data for Service Fitness Machine.
Discovered Characteristics FEA1, FEA2, and FEC9 for Service FEE7.
Characteristic FFF1 has no Descriptors.
Characteristic FFF2 has no Descriptors.
Characteristic FFF3 has no Descriptors.
Discovered Client Characteristic Configuration for Characteristic FFF4
Characteristic FFF5 has no Descriptors.
Characteristic Fitness Machine Feature has no Descriptors.
Discovered Client Characteristic Configuration for Characteristic Indoor Bike Data
Discovered Client Characteristic Configuration for Characteristic Treadmill Data
Discovered Client Characteristic Configuration for Characteristic Step Climber Data
Discovered Client Characteristic Configuration for Characteristic Cross Trainer Data
Discovered Client Characteristic Configuration for Characteristic Fitness Machine Control Point
Characteristic Supported Speed Range has no Descriptors.
Characteristic Supported Inclination Range has no Descriptors.
Characteristic Supported Resistance Level Range has no Descriptors.
Characteristic Supported Power Range has no Descriptors.
Characteristic Supported Heart Rate Range has no Descriptors.
Discovered Client Characteristic Configuration for Characteristic Fitness Machine Status
Discovered Client Characteristic Configuration for Characteristic Training Status
Discovered Client Characteristic Configuration for Characteristic Rower Data
Discovered Client Characteristic Configuration for Characteristic Stair Climber Data
Discovered Client Characteristic Configuration for Characteristic FEA1
Discovered Client Characteristic Configuration for Characteristic FEA2
Characteristic FEC9 has no Descriptors.



![IMG_0134](https://github.com/cagnulein/qdomyos-zwift/assets/162730216/8a34ed11-633e-44f1-a434-7113a7e2b1db)

**Reply by cagnulein** [reply]
yes it's compatible. if it doesn't work out of the box send me an email to roberto.viola83@gmail.com indicating this ticket and i will send you the beta

let me know

**Reply by MrPicklesRocks** [reply]
will do, thanks for the quick replies!
